tests/cppunit/types/json_test.cc

```diff
@@ -119,3 +119,21 @@ TEST_F(RedisJsonTest, Get) {
   ASSERT_TRUE(json_->Get(key_, {"$..x", "$..y", "$..z"}, &json_val_).ok());
   ASSERT_EQ(json_val_.Dump(), R"({"$..x":[{"y":1},4],"$..y":[[2,{"z":3}],1],"$..z":[{"a":{"x":4}},3]})");
 }
+
+TEST_F(RedisJsonTest, LegacyPath) {
+  ASSERT_THAT(json_->Set(key_, ".", "invalid").ToString(), MatchesRegex(".*syntax_error.*"));
+  ASSERT_THAT(json_->Set(key_, ".", "{").ToString(), MatchesRegex(".*Unexpected end of file.*"));
+
+  ASSERT_TRUE(json_->Set(key_, ".", "  \t{\n  }  ").ok());
+  ASSERT_TRUE(json_->Get(key_, {"."}, &json_val_).ok());
+  ASSERT_EQ(json_val_.Dump(), "[{}]");
+
+  auto s = json_->Set(key_, ".", "1");
+  ASSERT_TRUE(s.ok()) << s.ToString();
+  ASSERT_TRUE(json_->Get(key_, {"."}, &json_val_).ok());
+  ASSERT_EQ(json_val_.Dump(), "[1]");
+
+  ASSERT_TRUE(json_->Set(key_, ".", "[1, 2, 3]").ok());
+  ASSERT_TRUE(json_->Get(key_, {"."}, &json_val_).ok());
+  ASSERT_EQ(json_val_.Dump(), "[[1,2,3]]");
```

Comment:
Could you provide more tests about `tryConvertLegacyToJsonPath`?

Reply:
added, PTAL again